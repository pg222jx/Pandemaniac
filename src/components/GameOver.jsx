/**
 * Representing the 'Game over' status.
 *
 * @author Pernilla GÃ¶th
 * @version 1.0.0
 */


import React, { useEffect } from 'react'

const GameOver = (props) => {
    /**
    * Refreshes the page on Try Again.
    */
    const refreshPage = () => {
        window.location.reload(false)
    }

    /**
    * UseEffect hook to stop timer and hide Game components on render. 
    */
    useEffect(() => {
        props.onRender()
        props.stopTimer()
    })

    /**
    * Checks which virus was rendered to cause Game Over. 
    */
    const checkVirus = () => {
        switch (props.virus) {
            case 'NA':
                return 'North America'
            case 'SA':
                return 'South America'
            case 'EU':
                return 'Europe'
            case 'AF':
                return 'Africa'
            case 'AS':
                return 'Asia'
            case 'AU':
                return 'Australia'
            default:
        }
    }

    return (
        <div className="gameOver">
            <h3 id="gameOvertext">Game Over!</h3>
            <p data-testid="information_test">A fourth virus spawned in {checkVirus()}!</p><br/>
            <button className="tryAgain" onClick={refreshPage}>Try again?</button>
        </div>
    )
}
 
export default GameOver