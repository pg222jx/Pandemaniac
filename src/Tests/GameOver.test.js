import React from 'react'
import { shallow } from 'enzyme'
import GameOver from '../components/GameOver'

const wrapperShallow = shallow(<GameOver/>)


describe('testing the GameOver component', () => {
    it('should match the snapshot', () => {
        expect(wrapperShallow.html()).toMatchSnapshot()
      })
})
