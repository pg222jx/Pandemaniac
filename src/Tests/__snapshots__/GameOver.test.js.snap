// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the GameOver component should match the snapshot 1`] = `"<div class=\\"gameOver\\"><h3 id=\\"gameOvertext\\">Game Over!</h3><p data-testid=\\"information_test\\">A fourth virus spawned in !</p><br/><button class=\\"tryAgain\\">Try again?</button></div>"`;
